src/middleware/process.ts

```diff
@@ -21,8 +21,8 @@ export default async function processMiddleware(
     if (toCallMiddlewareIndex < middleware.length) {
       lastCalledMiddlewareIndex = toCallMiddlewareIndex;
       return middleware[toCallMiddlewareIndex]({
-        next: () => invokeMiddleware(toCallMiddlewareIndex + 1),
         ...initialArgs,
+        next: () => invokeMiddleware(toCallMiddlewareIndex + 1),
```

Comment:
🤔 To make [this middleware test pass](https://github.com/slackapi/bolt-js/pull/2139/files#diff-7a10c936267ab1a6a9b6eb4842b755717989a05090c641d56e81ace102e01a84R98-R106), yes. I also [think having middleware in this listener is the ideal behavior](https://github.com/slackapi/bolt-js/pull/2026#discussion_r1628868755)! This does feel like a risky change and I'm not sure what `...initialArgs` is overwriting...

It's unclear to me if these changes require this update or if this was an undercover bug beforehand... I'm leaning towards the undercover bug since `CustomFunction` acts as "global middleware" while things like `app.event` are "listener middleware" where this might not be noticed. I'm thinking the earlier removal of `next` from these arguments (also in `WorkflowStep`) served as a workaround to this limitation.

And IMO the distinction between listener and global middleware for these handlers is confusing but moving towards global middleware seems ideal for maintenance.